logstage/logstage-core/src/main/scala/izumi/logstage/macros/LoggerMacroMethods.scala

```diff
@@ -1,76 +1,107 @@
+    doLog(c)(message, Level.Error, EncodingMode.Strict)
   }
 
   def scCritMacroStrict(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
-    logMacro(c)(c.universe.reify(Log.Level.Crit), new LogMessageMacro0[c.type](c, strict = true).logMessageMacro(message), getEnclosingPosition(c))
+    doLog(c)(message, Level.Crit, EncodingMode.Strict)
   }
 
-  @inline private[this] def logMacro(
-    c: blackbox.Context { type PrefixType = AbstractLogger }
-  )(level: c.Expr[Log.Level],
-    message: c.Expr[Log.Message],
-    position: c.Expr[CodePositionMaterializer],
-  ): c.Expr[Unit] = {
+  def scTraceMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Trace, EncodingMode.Raw)
+  }
+
+  def scDebugMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Debug, EncodingMode.Raw)
+  }
+
+  def scInfoMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Info, EncodingMode.Raw)
+  }
+
+  def scWarnMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Warn, EncodingMode.Raw)
+  }
+
+  def scErrorMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Error, EncodingMode.Raw)
+  }
+
+  def scCritMacroRaw(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String]): c.Expr[Unit] = {
+    doLog(c)(message, Level.Crit, EncodingMode.Raw)
+  }
+
+  @inline private[this] def doLog(c: blackbox.Context { type PrefixType = AbstractLogger })(message: c.Expr[String], level: Level, mode: EncodingMode): c.Expr[Unit] = {
+    val m: c.Expr[Message] = mode.fold(c.universe.reify(Message.raw(message.splice))) {
+      strict =>
+        new LogMessageMacro0[c.type](c, strict = strict).logMessageMacro(message)
+    }
+    val l = level match {
+      case Level.Trace =>
+        c.universe.reify(Level.Trace)
+      case Level.Debug =>
+        c.universe.reify(Level.Debug)
+      case Level.Info =>
+        c.universe.reify(Level.Info)
+      case Level.Warn =>
+        c.universe.reify(Level.Warn)
+      case Level.Error =>
+        c.universe.reify(Level.Error)
+      case Level.Crit =>
+        c.universe.reify(Level.Crit)
+    }
+
     c.universe.reify {
-      {
-        val self = c.prefix.splice
-        if (self.acceptable(Log.LoggerId(CodePositionMaterializerMacro.getApplicationPointId(c).splice), level.splice)) {
-          self.unsafeLog(Log.Entry.create(level.splice, message.splice)(position.splice))
-        }
-      }
+      c.prefix.splice.log(l.splice)(m.splice)(getEnclosingPosition(c).splice)
```

Comment:
@catostrophe Ergh, just noticed this, this change degrades performance (inline if is faster than a call)

Reply:
Ouch. Sorry about that. Can't even remember why I touched it...

Reply:
No worries